sdk/src/main/java/com/mapbox/maps/NativeObserver.kt

```diff
@@ -0,0 +1,441 @@
+package com.mapbox.maps
+
+import android.os.Handler
+import com.mapbox.maps.extension.observable.*
+import com.mapbox.maps.plugin.delegates.listeners.*
+import java.lang.ref.WeakReference
+import java.util.concurrent.CopyOnWriteArrayList
+
+internal class NativeObserver(
```

Comment:
When registering or unregistering, we are using a handler to post to make sure that happens on the main thread. Imo calls on the NativeObserver should be default be on the main thread. (that callbacks get invoked on another, is independent of this).

Reply:
This is added since we introduced the on demand subscription.
Since the events will be emitted on the same thread the thread that subscribe is called, I'm thinking there would be a case that user firstly adds a listener on a non-main thread and then in some other place adds a second listener on the main thread and expect the events is emitted on the main thread, but then that is not the case, since the event is still emitted to the thread that the first subscriber's thread during subscription.
To be consistent I think it would make sense to always call the subscribe on the main thread, wdyt?

Reply:
:+1: on only main thread requirement, @alexshalamov confirmed this.

Reply:
@pengdev just to make sure I get it all right - we use handler post for only registering / unregistering listeners while `notify` is always called from main thread and concrete listeners get notified immediately on main thread (and do not post that updates on main thread handler).

Reply:
@kiryldz right, that is how it is expected to work.